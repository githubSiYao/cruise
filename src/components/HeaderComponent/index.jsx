import React from 'react'
import { Link } from 'react-router-dom'
import './index.scss'
import logo from '../../logo.svg'

export default class HeaderComponent extends React.PureComponent {
  constructor(props) {
    super(props)

    // define user menu list data
    this.menu = [
      { icon: 'icon-id-card', label: 'Profile' },
      { icon: 'icon-sign-in', label: 'Sign Out' }
    ]
    this._renderUserMenuList = this._renderUserMenuList.bind(this)
  }

  // render user menu list
  _renderUserMenuList() {
    return (
      <div className="header__user-menu-area">
        <div className="header__user-avatar">
          <span className="header__user-avatar-img"></span>
          <span className="iconfont icon-angle-down"></span>
        </div>
        <ul className="header__user-menu-list">
          {this.menu.map((item, index) => (
            <li className="header__user-menu-list-item" key={index}>
              <span className={`iconfont ${item.icon}`}></span>
              <span className="header__user-menu-list-item-label">{item.label}</span>
            </li>
          ))}
        </ul>
      </div>
    )
  }

  render() {
    return (
      <header className="header">
        <div className="header__wrapper">
          <Link to="/" className="header__logo">
            <img src={logo} alt="Cruise" />
          </Link>
          {this._renderUserMenuList()}
        </div>
      </header>
    )
  }
}