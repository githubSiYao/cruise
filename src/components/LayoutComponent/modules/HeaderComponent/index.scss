@import "../../../../scss/variables.scss";

.header {
  position: fixed;
  top: 0;
  z-index: 99;
  width: 100%;
  background-color: #ffffff;
  box-shadow: 1px 4px 10px rgba(0, 0, 0, 0.1);

  &__wrapper {
    position: relative;
    max-width: 1200px;
    margin: 0 auto;
    display: flex;
    justify-content: space-between;
  }

  &__logo {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }

  &__logo img {
    height: 40px;
    width: auto;
  }

  &__user-menu-area {
    position: relative;
    padding: 12px;
    margin-right: 10px;
    display: inline-flex;
    align-items: center;

    &:hover .header__user-menu-list,
    .header__user-menu-list:hover {
      display: block;
    }
  }

  &__user-avatar {
    &-img {
      display: inline-block;
      width: 40px;
      height: 40px;
      margin-right: 10px;
      border-radius: 50%;
      vertical-align: middle;
      background: center no-repeat;
      background-color: #dddddd;
      background-image: url(/assets/mockups/1/Guide.png);
      background-position-x: -625px;
      background-position-y: -555px;
    }

    .iconfont {
      color: #435466;
    }
  }

  &__user-menu-list {
    position: absolute;
    right: 0;
    top: 100%;
    display: none;
    padding: 10px 0;
    background-color: #ffffff;
    box-shadow: 1px 4px 10px rgba(0, 0, 0, 0.1);

    &-item {
      padding: 8px 12px;
      color: #2d4054;
      display: flex;
      align-items: center;
      min-width: 128px;
      cursor: pointer;
      span.iconfont {
        width: 20px;
        font-size: 14px;
      }
      &:hover {
        background-color: #efefef;
      }

      &-label {
        padding-left: 10px;
      }
    }
  }

  &__nav-menu {
    @media screen and (max-width : 1024px){
      visibility: visible;
      cursor: pointer;
    }
    visibility: hidden;
    padding: 22px 15px;
    .iconfont {
      color: #435466;
    }
  }
}
