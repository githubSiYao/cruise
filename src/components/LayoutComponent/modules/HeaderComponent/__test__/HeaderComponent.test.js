import React from 'react'
import { StaticRouter } from 'react-router-dom'
import TestRender from 'react-test-renderer'
import HeaderComponent from '../index'

describe('<HeaderComponent />', () => {
  it('HeaderComponent Snapshot', () => {
    const testRender = TestRender.create(
      <StaticRouter>
        <HeaderComponent />
      </StaticRouter>
    )
    expect(testRender.toJSON()).toMatchSnapshot()
  })
})