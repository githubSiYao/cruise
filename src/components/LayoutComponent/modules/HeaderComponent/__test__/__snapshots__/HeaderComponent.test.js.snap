// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderComponent /> HeaderComponent Snapshot 1`] = `
<header
  className="header"
>
  <div
    className="header__wrapper"
  >
    <a
      className="header__logo"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Cruise"
        src="logo.svg"
      />
    </a>
    <div
      className="header__user-menu-area"
    >
      <div
        className="header__user-avatar"
      >
        <span
          className="header__user-avatar-img"
        />
        <span
          className="iconfont icon-angle-down"
        />
      </div>
      <ul
        className="header__user-menu-list"
      >
        <li
          className="header__user-menu-list-item"
        >
          <span
            className="iconfont icon-id-card"
          />
          <span
            className="header__user-menu-list-item-label"
          >
            Profile
          </span>
        </li>
        <li
          className="header__user-menu-list-item"
        >
          <span
            className="iconfont icon-sign-in"
          />
          <span
            className="header__user-menu-list-item-label"
          >
            Sign Out
          </span>
        </li>
      </ul>
    </div>
  </div>
</header>
`;
